### DISCUSSION TITLE
Code changes not picked up after full clean... Compile cache lingering?

### DISCUSSION THREAD

**Discussion by JpnTrzdTzURv3lOIQad0** [PlummersSoftwareLLC/NightDriverStrip#608] [Q&A] [answered]
This is hopefully an easy question.  
After a successful compile/upload to a M5StickC-Plus, I remapped LED_PIN0 to 26 (globals.h line 282), but after a pio full clean / build / upload the firmware still uses the stock G32 pin. A similar issue occurred with a separate Heltec v3 where secrets.h wifi changes were not getting picked up after rebuilding.  Is there a cache output path that requires a manual clean step?  
IDE is a stock VS code install with PlatformIO.

**Comment by robertlipe** [comment]
Did we just have this discussion a few days ago, starting somewhere around:
https://github.com/PlummersSoftwareLLC/NightDriverStrip/issues/598#issuecomment-1912944707
...which, itself, was later deemed to be a duplicate of yet another "WTH"
bag of mysteries.

In those two issues, two of us concluded that OTAs for development are
absolutely not worth the time it saves you plugging in a cable becaues you
can get stuck in mysterious cases where you can change the code, but
eventually, even if a cable is involved, it simply runs some old version,
presumably in the other OTA partition.

So before we go into details:
1) Are you now or have you ever used OTA to shoot new code into the board?
2) Does totally vaporizing your flash memory (my working theory is it's the
parition table that's to blame, but Im not debugging this at the byte level
unless this problem gets a LOT worse) using the recipies described around
https://github.com/PlummersSoftwareLLC/NightDriverStrip/issues/598#issuecomment-1912948038
restore your device to a state where it works 100%?

My working theory is that something in the ota update path doesn't tell the
device to change the partition to AND USE the newly downloaded image and I
suspect that it gets stuck if the active partition is the second one.

From a human frustration point, by the time you know you're affected,
you're already pulling your hair out and doubting your sanity, so dumping
partition tables and doing binary dumps of the two partitions and figuring
out where the "acctive" flag is and all that sounds like about as much fun
as closing your finger in a car door and recognizing the keys are locked
inside the car.

**Comment by heidepiek** [comment, marked as answer]
![m5stick c plus ledpin](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/bd1cc50a-4b41-4311-bf52-697c3804a96d)

It says #define LED_PIN0 32 twice.
Have now defined both to 26, and voilla the led strip works on pin 26.


https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/a2aa8fa0-d887-4a8a-9e17-fd19b836f286



Greetz harrie

oh yeah
demo is i (think) not used for m5stick c plus
m5demo is for the m5stick c plus

u said (I remapped LED_PIN0 to 26 (globals.h line 282), but this is for the demo, not m5 demo (i think)

**Comment by JpnTrzdTzURv3lOIQad0** [comment]
Thank you both for the quick responses. The Heltec issue was due to my inexperience with Platform IO. This has been a fun project to use, and the kids love the sound responsive effects!
